### PR TITLE
task-NXDOC-2546-migration-management-endpoint-documentation

### PR DESCRIPTION
**Pull request by NourNuxeo**

### CODE REVIEW

File: src/rest-api/endpoints/management-endpoint/migration-endpoint.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+### Path Parameters
+
+| Parameter Name   | Type       | Description      |
+| ---------------- | ---------- | ---------------- |
+| **MIGRATION_ID** | **string** | The migration id.|
+
+### Response
+
+If successful, probes the current state of a migration by analyzing persistent data, and sets it as the new current state, then returns a JSON representation of the requested migration object.
```
Comment (ataillefer) [inline src/rest-api/endpoints/management-endpoint/migration-endpoint.md:126]:
```suggestion
If successful, probes the current state of a migration by analyzing persistent data, sets it as the new current state, then returns a JSON representation of the requested migration object.
```
Though, not sure what "sets it as the new current state" means.

File: src/rest-api/endpoints/management-endpoint/migration-endpoint.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+| **MIGRATION_ID** | **string** | The migration id.|
+
+### Response
+
+If successful, probes the current state of a migration by analyzing persistent data, sets it as the new current state, then runs the migration asynchronously.
+
+### Status Codes
+
+- 202 _Accepted_ - Success.
+- 400 _Bad Request_ - Unknown migration `MIGRATION_ID`.
```
Comment (ataillefer) [inline src/rest-api/endpoints/management-endpoint/migration-endpoint.md:179]: Isn't it 404 like for the other calls?

File: src/rest-api/endpoints/management-endpoint/migration-endpoint.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+| **MIGRATION_STEP_ID** | **string** | The migration step id.|
+
+### Response
+
+If successful, runs the migration step asynchronously.
+
+### Status Codes
+
+- 202 _Accepted_ - Success.
+- 400 _Bad Request_ - Unknown migration for `MIGRATION_ID` OR Unknown migration step `MIGRATION_STEP_ID` for `MIGRATION_ID`.
```
Comment (ataillefer) [inline src/rest-api/endpoints/management-endpoint/migration-endpoint.md:208]: Same remark, isn't it a 404?
And keep the same description as fr the others? "Migration with the given `MIGRATION_ID` or migration step with the given `MIGRATION_STEP_ID` does not exist for `MIGRATION_ID`."

Comment (NourNuxeo) [reply src/rest-api/endpoints/management-endpoint/migration-endpoint.md:208]: Same here with the 2 variations if the `MIGRATION_ID` or the `MIGRATION_STEP_ID` is wrong.

File: src/rest-api/endpoints/management-endpoint/migration-endpoint.md (hunk 1 above)
Comment (NourNuxeo) [reply src/rest-api/endpoints/management-endpoint/migration-endpoint.md:126]: Ok for removing the "and"
About the new current state. Here I'm quoting the Javadoc:
https://github.com/nuxeo/nuxeo-lts/blob/2021/modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationService.java#L236

File: src/rest-api/endpoints/management-endpoint/migration-endpoint.md (hunk 2 above)
Comment (ataillefer) [reply src/rest-api/endpoints/management-endpoint/migration-endpoint.md:179]: OK, but still, I believe that it is clearer for the reader to always have the same phrase, no matter the internal message sent by the Nuxeo server.
Indeed, in the "Get Migration Object" section we have:
```
### Status Codes

- 200 _OK_ - Success.
- 404 _Not Found_ - Migration with the given `MIGRATION_ID` does not exist.
```
Let's use the exact same format for the current section "Run a Migration"

File: src/rest-api/endpoints/management-endpoint/migration-endpoint.md (hunk 3 above)
Comment (NourNuxeo) [reply src/rest-api/endpoints/management-endpoint/migration-endpoint.md:208]: Migration with the given `MIGRATION_ID` or `MIGRATION_STEP_ID` does not exist.

**Review by ataillefer — APPROVED** [review]

**Pull request merged**